src/challenges.js

```diff
@@ -30,6 +54,10 @@ const cacheFunction = cb => {
 const reverseStr = str => {
   // reverse str takes in a string and returns that string in reversed order
   // The only difference between the way you've solved this before and now is that you need to do it recursivley!
+  if (str === '') {
+    return;
+  }
+  return reverseStr(str.substr(1)) + str.charAt(0);
 };
 
 const checkMatchingLeaves = obj => {
```

Comment:
Come back soon and play around with this one. It's a big problem. But I think you could do it now! :)